### PR TITLE
[RNMobile] - Blog layout template

### PR DESCRIPTION
**Pull request by geriux**
`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2114

## Description
This PR adds the `Blog` layout since the `Latest Posts` block is now available.

## Screenshots

<details><summary>Blog Layout Preview</summary>

Blog preview Android | Blog preview Android
-|-
 | 

Blog preview iOS | Blog preview iOS
-|-
 | 

</details>

<details><summary>Emoji compatibility check</summary>

### Android

API 21 | API 23
-|-
 |

API 27 | API 29
-|-
 |

### iOS
 

</details>

## How has this been tested?
- Open the WordPress app with metro running
- Go to Pages
- Add a new page
- **Expect** to see the blog option in the layout picker, tap on it
- **Expect** to see the preview of the template
- Tap on apply
- **Expect** to see the layout applied in the editor
- Preview the page
- **Expect** to see the layout showing correctly

## Types of changes
New layout template

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by mkevins** [comment]
Tested on Android (Pixel 3a) and iOS (iPhone 11), and it works as described. One thing I noticed (also visible in your screenshots) is that the horizonal alignment is not always consistent. E.g. on Android "Follow our Blog" is left-aligned, while on iOS it is center-aligned.

However, on the _page_ preview, both are centered:

Android | iOS
-|-
blog-template-page-preview-android | blog-template-page-preview-ios

File: packages/block-editor/src/components/page-template-picker/templates/blog.native.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				content: __( 'Follow our Blog' ),
+				level: 2,
+			},
+		},
+		{
+			name: 'core/button',
+			attributes: {
+				// translators: sample content for "Blog" page template
+				text: __( 'Subscribe' ),
+				align: 'center',
```
Comment (mkevins) [inline packages/block-editor/src/components/page-template-picker/templates/blog.native.js:67]: I see that button should also be centered, but instead it seems to render left aligned. I didn't see the `align` attribute in the button [block.json](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/button/block.json) Do you know offhand if this is / will be a supported attribute of the button block?

**Comment by geriux** [comment]
Hey @mkevins ! 👋  Thanks for reviewing!

> Tested on Android (Pixel 3a) and iOS (iPhone 11), and it works as described. One thing I noticed (also visible in your screenshots) is that the horizonal alignment is not always consistent. E.g. on Android "Follow our Blog" is left-aligned, while on iOS it is center-aligned.

Nice catch =) so there's [an issue on Android](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1880#issuecomment-604095497) where we don't support alignments yet, there's work on it in progress though. On iOS it renders correctly.

File: packages/block-editor/src/components/page-template-picker/templates/blog.native.js (hunk 1 above)
Comment (geriux) [reply packages/block-editor/src/components/page-template-picker/templates/blog.native.js:67]: Oh yeah, we don't support that and we are working on `Buttons` block which is a wrapper that has `Button` in it and adds alignment capabilities. I'll remove the attribute.

Comment (geriux) [reply packages/block-editor/src/components/page-template-picker/templates/blog.native.js]: https://github.com/WordPress/gutenberg/pull/21426/commits/9e4ece6fe7d96f9bed6dbf5da1657c1b3306de24

**Review by mkevins — APPROVED** [review]
Approved with https://github.com/WordPress/gutenberg/commit/9e4ece6fe7d96f9bed6dbf5da1657c1b3306de24. Nice work!

**Pull request merged**